packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -172,6 +183,27 @@ export class TabBarRenderer extends TabBar.Renderer {
         return h.div({ className: 'p-TabBar-tabLabel', style }, data.title.label);
     }
 
+    protected getDecorations(tab: string): WidgetDecoration.Data[] {
+        const tabDecorations = [];
+        if (this.tabBar && this.decoratorService) {
+            const allDecorations = this.decoratorService.getDecorations([...this.tabBar.titles]);
+            if (allDecorations.has(tab)) {
+                tabDecorations.push(...allDecorations.get(tab));
+            }
+        }
+        return tabDecorations;
+    }
+
+    protected getDecorationData<K extends keyof WidgetDecoration.Data>(tab: string, key: K): WidgetDecoration.Data[K][] {
+        return this.getDecorations(tab).filter(data => data[key] !== undefined).map(data => data[key]);
+
+    }
+
+    private getIconClass(iconName: string | string[], additionalClasses: string[] = []): string {
+        const iconClass = (typeof iconName === 'string') ? ['a', 'fa', `fa-${iconName}`] : ['a'].concat(iconName);
+        return iconClass.concat(additionalClasses).join(' ');
+    }
+
     /**
      * Find duplicate labels from the currently opened tabs in the tab bar.
      * Return the approriate partial paths that can distinguish the identical labels.
```

Comment:
Your previous PR ##5775 introduced this typo, can you fix it since you are modifying this file already?
`approriate` should be `appropriate`.